### PR TITLE
Bar from space type ratios

### PR DESCRIPTION
**Pull request by DavidGoldwasser**
In addition to new methods this updates the rake spec to test create_bar methods. I need to this to test other pull requests

### CODE REVIEW

File: lib/openstudio/extension/core/os_lib_model_generation.rb [hunk 1]
```diff
@@ -1718,40 +1719,16 @@ def bar_from_building_type_ratios(model, runner, user_arguments)
     if !one_or_greater then return false end
```
Comment (nllong) [inline lib/openstudio/extension/core/os_lib_model_generation.rb:1719]:
```suggestion
    return false if !one_or_greater
```

File: lib/openstudio/extension/core/os_lib_model_generation.rb [hunk 2]
```diff
@@ -1718,40 +1719,16 @@ def bar_from_building_type_ratios(model, runner, user_arguments)
...
     if !non_neg then return false end
```
Comment (nllong) [inline lib/openstudio/extension/core/os_lib_model_generation.rb:1720]:
```suggestion
    return false if !non_neg
```

File: openstudio-extension.gemspec [hunk 3]
```diff
@@ -33,6 +33,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'octokit', '~> 4.18.0' # for change logs
   spec.add_dependency 'openstudio-workflow', '~> 2.1.1'
   spec.add_dependency 'openstudio_measure_tester', '~> 0.2.3'
+  spec.add_dependency 'openstudio-standards', '~> 0.2.12' # for os_lib unit tests
```
Comment (nllong) [inline openstudio-extension.gemspec:36]: This wasn't in there before, so why is this needed now?

File: spec/openstudio/core/os_lib_model_generation_spec.rb [hunk 4]
```diff
@@ -0,0 +1,234 @@
+# *******************************************************************************
+# OpenStudio(R), Copyright (c) 2008-2020, Alliance for Sustainable Energy, LLC.
```
Comment (nllong) [inline spec/openstudio/core/os_lib_model_generation_spec.rb:2]: 2021

File: spec/openstudio/core/os_lib_model_generation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+require 'openstudio/extension/core/os_lib_helper_methods'
+require 'openstudio/extension/core/os_lib_geometry'
+require 'openstudio/extension/core/os_lib_model_generation'
+require 'openstudio/extension/core/os_lib_model_simplification'
+require 'openstudio-standards'
+
+# adding this because I may want to inspect runner output
+require 'openstudio/measure/ShowRunnerOutput'
+
+RSpec.describe 'Bar Methods' do # include from building type ratios, space type ratios, and from building
```
Comment (nllong) [inline spec/openstudio/core/os_lib_model_generation_spec.rb:45]: NICE! TESTS!

**Review by nllong — COMMENTED** [review]
Looks great, couple comments...

File: openstudio-extension.gemspec (hunk 3 above)
Comment (DavidGoldwasser) [reply openstudio-extension.gemspec:36]: Because the new spec tests exercise methods in the extension gem that use standards. It failed to find standards without this.

Comment (nllong) [reply openstudio-extension.gemspec:36]: then perhaps we make this `add_development_dependency`?

File: spec/openstudio/core/os_lib_model_generation_spec.rb (hunk 5 above)
Comment (DavidGoldwasser) [reply spec/openstudio/core/os_lib_model_generation_spec.rb:45]: Thanks, I'll try to add tests for any code I work on from now on. I'll also add test for create_typical method very soon. That will be a little slower because it does sizing run.

File: spec/openstudio/core/os_lib_model_generation_spec.rb (hunk 4 above)
Comment (DavidGoldwasser) [reply spec/openstudio/core/os_lib_model_generation_spec.rb]: I just made a commit for this

File: openstudio-extension.gemspec (hunk 3 above)
Comment (nllong) [reply openstudio-extension.gemspec:36]: I think we need to discuss how this is going to work. The extension gem shouldn't have this heavy of a dependency. Can we move all of this into the model articulation gem?

Comment (nllong) [inline openstudio-extension.gemspec:36]:
```suggestion
  spec. add_development_dependency 'openstudio-standards', '~> 0.2.12' # for os_lib unit tests
```
